frappe/public/js/frappe/router.js

```diff
@@ -39,13 +39,20 @@ frappe.route = function() {
 
 	frappe.route_history.push(route);
 
-	if(route[0] && route[1] && frappe.views[route[0] + "Factory"]) {
-		// has a view generator, generate!
-		if(!frappe.view_factory[route[0]]) {
-			frappe.view_factory[route[0]] = new frappe.views[route[0] + "Factory"]();
-		}
+	if(route[0]) {
+		const title_cased_route = frappe.utils.to_title_case(route[0]);
+
+		if(route[1] && frappe.views[title_cased_route + "Factory"]) {
+			// has a view generator, generate!
+			if(!frappe.view_factory[title_cased_route]) {
+				frappe.view_factory[title_cased_route] = new frappe.views[title_cased_route + "Factory"]();
+			}
 
-		frappe.view_factory[route[0]].show();
+			frappe.view_factory[title_cased_route].show();
+		} else {
+			// show page
+			frappe.views.pageview.show(route[0]);
+		}
 	} else {
 		// show page
 		const route_name = frappe.utils.xss_sanitise(route[0]);
```

Comment:
I think this line should be moved above line 54.
since passing nothing (because `route[0]` is null here) to `frappe.utils.xss_sanitise()` will raise error.